src/Session.js

```diff
@@ -1520,8 +1520,12 @@ InviteClientContext.prototype = {
               (this.earlyDialogs[id].pracked[this.earlyDialogs[id].pracked.length-1] >= response.getHeader('rseq') && this.earlyDialogs[id].pracked.length > 0)) {
             return;
           }
+
           // TODO: This may be broken. It may have to be on the early dialog
-          this.sessionDescriptionHandler = this.sessionDescriptionHandlerFactory(this, this.sessionDescriptionHandlerFactoryOptions);
+          if (!this.ua.configuration.allowEarlyMedia) {
```

Comment:
I do not think that this block of code is necessary. I think it is best to revert it to how it was previously.

Reply:
Must say I am confused, there is only one way `this.sessionDescriptionHandler` is going to be `null` at this point which is an INVITE without SDP. Creating a new sessionDescriptionHandler is going to initiate a new peerConnection, different from the one used in my INVITE request no?

Reply:
What I am saying is that the block is not necessary and potentially sets up an undefined variable error on line 1529 below. It is possible to get to this code and have the allowEarlyMedia flag be `true`. That would cause the SDH to not get instantiated and then an undefined error to spit out on the next `if` statement.

Reply:
I see, then an `if` statement checking if `this.sessionDescriptionHandler` is defined should do the trick right ?

Reply:
Yes. That would work fine.